src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -239,29 +261,33 @@ UCollator* CloneCollatorWithOptions(const UCollator* pCollator, int32_t options,
     }
 
     UCollator* pClonedCollator;
-    std::vector<UChar> customRules = GetCustomRules(options, strength, isIgnoreSymbols);
-    if (customRules.empty())
+    UCharList* customRules = GetCustomRules(options, strength, isIgnoreSymbols);
```

Comment:
>UCharList* customRules = GetCustomRules(options, strength, isIgnoreSymbols); [](start = 4, length = 76)

This maybe ok but I am wondering if this is kind of behavior change? I mean before if we had out of memory exception in C++ we wouldn't continue coming to this code. but now we are kind of ignoring the out of memory and just continue without using any of custom rules.
I just thinking loud what is the best we can do here? should we ignore the OOM or we should fail the whole call?

Reply:
I think we can change the signature from:

```c
UCharList*      GetCustomRules(int32_t, UColAttributeValue, bool)
```

to:

```c
RESULT_CODE     GetCustomRules(int32_t, UColAttributeValue, bool, UCharList**)
```

and return `U_MEMORY_ALLOCATION_ERROR` like we are doing elsewhere; including this file.

Reply:
This will be reasonable